src/DateTimeParser.jl

```diff
@@ -154,23 +150,24 @@ function _parsedate(datetimestring::AbstractString; fuzzy::Bool=false,
             if length(ymd) == 3 && tokenlength in (2,4) &&
                 (i>=len || (tokens[i] != ":" && !haskey(HMS, lowercase(tokens[i]))))
                 # 19990101T23[59]
-                res["hour"] = parse(Int, token[1:2])
+                res["hour"] = Hour(token[1:2])
                 if tokenlength == 4
-                    res["minute"] = parse(Int, token[3:4])
+                    res["minute"] = Minute(token[3:4])
                 end
             elseif tokenlength == 6
                 # YYMMDD or HHMMSS[.ss]
                 if length(ymd) != 0 || (i+1 <= len && tokens[i] == "." && isdigit(tokens[i+1]))
                     # 19990101T235959[.59]
-                    res["hour"] = parse(Int, token[1:2])
-                    res["minute"] = parse(Int, token[3:4])
-                    res["second"] = parse(Int, token[5:6])
+                    res["hour"] = Hour(token[1:2])
+                    res["minute"] = Minute(token[3:4])
+                    res["second"] = Second(token[5:6])
                     if i+1 <= len && tokens[i] == "." && isdigit(tokens[i+1])
-                        res["millisecond"] = round(Int, 1000 * parse(Float64, string(tokens[i], tokens[i+1])))
+                        temp = round(Int, 1000 * parse(Float64, string(tokens[i], tokens[i+1])))
+                        res["millisecond"] = Millisecond(temp)
                         i += 2
                     end
                 else
-                    push!(ymd, convertyear(parse(Int, token[1:2])))
+                    push!(ymd, convertyear(Year(token[1:2])).value)
```

Comment:
If there are no indicators of what the 6 digit number is then we assume it is YYMMDD. I agree this is kind of arbitrary. But unlike the YY-MM-DD case this could also have a 4 digit year or be HHMMSS. This is the way the python parser works. It would also make sense to parse it into three 2 digit tokens for ymd, or a 4 digit token and a 2 digit token, or a 2 digit token and a 4 digit token, or HHMMSS.